Import the data

In [50]:
import pandas as pd

df = pd.read_csv("COVID-data_consolidated - County_City Race.csv", header = 0)
df.describe()
df.head()
# df

date   county       city           race  population  atleast_onedose  \
0      3/9/21  Suffolk    Chelsea          AI/AN        53.0              0.0   
1      3/9/21  Suffolk    Chelsea          Asian       810.0            207.0   
2      3/9/21  Suffolk    Chelsea          Black      3336.0            399.0   
3      3/9/21  Suffolk    Chelsea       Hispanic     25092.0           1791.0   
4      3/9/21  Suffolk    Chelsea          Multi       420.0             81.0   
...       ...      ...        ...            ...         ...              ...   
2983  10/4/22  Norfolk  Wellesley          Multi      1144.0           1535.0   
2984  10/4/22  Norfolk  Wellesley          NH/PI        10.0              0.0   
2985  10/4/22  Norfolk  Wellesley          White     21875.0          20020.0   
2986  10/4/22  Norfolk  Wellesley  Other/Unknown         NaN              0.0   
2987  10/4/22  Norfolk  Wellesley          Total     29550.0          27970.0   

      fully_vaccinated  boosted  boosted_second  partially_vaccinated  
0                  0.0      NaN             NaN                   0.0  
1                109.0      NaN             NaN                  98.0  
2                237.0      NaN             NaN                 162.0  
3                810.0      NaN             NaN                 981.0  
4                 50.0      NaN             NaN                  31.0  
...                ...      ...             ...                   ...  
2983            1457.0   1125.0           363.0                  78.0  
2984               0.0      0.0             0.0                   0.0  
2985           17764.0  13048.0          5015.0                2256.0  
2986               0.0      0.0             0.0                   0.0  
2987           24061.0  16986.0          5971.0                3909.0  

[2988 rows x 10 columns]

In [49]:
df_chelsea = df[df['city'] =='Chelsea']
df_newton = df[df['city'] =='Newton']
df_revere = df[df['city'] =='Revere']
df_wellesley = df[df['city'] == 'Wellesley']

Hypothesis Testing Between Newton and Chelsea:

Hypothesis1:
H0: The mean of fully vaccinated for black people in Chelsea is the same as that in Newton.
Ha: The mean of fully vaccinated for black people in Chelsea is different from that in Newton.

In [64]:
# Example of the Student's t-test
from scipy.stats import ttest_ind
# Example of the Analysis of Variance Test
from scipy.stats import f_oneway

df_chelsea_black = df_chelsea[df_chelsea['race'] == 'Black']
df_newton_black = df_newton[df_newton['race'] == 'Black']

def t_test(df1, df2):
	stat,p = ttest_ind(df1, df2 ,equal_var = False)
	print('stat=%.3f, p=%.8f' % (stat, p))
	if p > 0.05:
		print('Probably the same distribution')
	else:
		print('Probably different distributions')

	return p

t_test(df_newton_black['fully_vaccinated'], df_chelsea_black['fully_vaccinated'])

stat=4.889, p=0.00000240
Probably different distributions


2.4021242392806963e-06

In [47]:
# df_chelsea_white = df_chelsea[df_chelsea['race'] == 'White']
# df_newton_white= df_newton[df_newton['race'] == 'White']
# t_test(df_newton_white['fully_vaccinated'], df_chelsea_white['fully_vaccinated'])

stat=36.057, p=0.00000000
Probably different distributions


In [48]:
# df_chelsea_hispanic = df_chelsea[df_chelsea['race'] == 'Hispanic']
# df_newton_hispanic= df_newton[df_newton['race'] == 'Hispanic']
# t_test(df_newton_hispanic['fully_vaccinated'], df_chelsea_hispanic['fully_vaccinated'])

stat=-20.504, p=0.00000000
Probably different distributions


In [74]:
city = ['Chelsea', 'Newton', 'Revere', 'Wellesley']
race = ['Black','Hispanic','White']
def hypothesis_test(df):
	for i in range(len(city)-1):
		for j in range(i+1, len(city)):
			df1 = df[df['city'] == city[i]]
			df2 = df[df['city'] == city[j]]
			print(city[i], 'vs.', city[j])
			for a in range(len(race)):
				df_1 = df1[df1['race'] == race[a]]
				df_2 = df2[df2['race'] == race[a]]
				print(race[a])
				t_test(df_1['fully_vaccinated'], df_2['fully_vaccinated'])

			print('\n')

hypothesis_test(df)


Chelsea vs. Newton
Black
stat=-4.889, p=0.00000240
Probably different distributions
Hispanic
stat=20.504, p=0.00000000
Probably different distributions
White
stat=-36.057, p=0.00000000
Probably different distributions


Chelsea vs. Revere
Black
stat=-3.468, p=0.00067271
Probably different distributions
Hispanic
stat=1.391, p=0.16610254
Probably the same distribution
White
stat=-25.046, p=0.00000000
Probably different distributions


Chelsea vs. Wellesley
Black
stat=23.799, p=0.00000000
Probably different distributions
Hispanic
stat=25.192, p=0.00000000
Probably different distributions
White
stat=-21.231, p=0.00000000
Probably different distributions


Newton vs. Revere
Black
stat=1.134, p=0.25838889
Probably the same distribution
Hispanic
stat=-20.588, p=0.00000000
Probably different distributions
White
stat=24.964, p=0.00000000
Probably different distributions


Newton vs. Wellesley
Black
stat=29.140, p=0.00000000
Probably different distributions
Hispanic
stat=23.603, p=0.00000000
Pro

# Summary:

## Chelsea vs. Newton

### 1. Black People

H0: μ1 = μ2
Ha: μ1 != μ2

stat = -4.889
p = 0.00000240

Conclusion: Since the p-value is small enough, we can reject the null hypothesis and conclude that the mean of fully vaccinated for black people in Chelsea is significantly different from that in Newton.

### 2. Hispanic People

H0: μ1 = μ2
Ha: μ1 != μ2

stat = 20.504
p = 0.00000000

Conclusion: Since the p-value is small enough, we can reject the null hypothesis and conclude that the mean of fully vaccinated for hispanic people in Chelsea is significantly different from that in Newton.

### 3. White People

H0: μ1 = μ2
Ha: μ1 != μ2

stat = -36.057
p = 0.00000000
Probably different distributions

Conclusion: Since the p-value is small enough, we can reject the null hypothesis and conclude that the mean of fully vaccinated for white people in Chelsea is significantly different from that in Newton.



## Chelsea vs. Revere

### 1. Black People

H0: μ1 = μ2
Ha: μ1 != μ2

stat = -3.468
p = 0.00067271

Conclusion: Since the p-value is small enough, we can reject the null hypothesis and conclude that the mean of fully vaccinated for black people in Chelsea is significantly different from that in Revere.

### 2. Hispanic People

H0: μ1 = μ2
Ha: μ1 != μ2

stat = 1.391
p = 0.16610254

Conclusion: Since the p-value is greater than the significance level, we fail to reject the null hypothesis and conclude that the mean of fully vaccinated for hispanic people in Chelsea is NOT significantly different from that in Revere.

### 3. White People

H0: μ1 = μ2
Ha: μ1 != μ2

stat = -25.046
p = 0.00000000
Probably different distributions

Conclusion: Since the p-value is small enough, we can reject the null hypothesis and conclude that the mean of fully vaccinated for white people in Chelsea is significantly different from that in Revere.


## Chelsea vs. Wellesley

### 1. Black People

H0: μ1 = μ2
Ha: μ1 != μ2

stat = 23.799
p = 0.00000000

Conclusion: Since the p-value is small enough, we can reject the null hypothesis and conclude that the mean of fully vaccinated for black people in Chelsea is significantly different from that in Wellesley.

### 2. Hispanic People

H0: μ1 = μ2
Ha: μ1 != μ2

stat = 25.192
p = 0.00000000

Conclusion: Since the p-value is small enough, we can reject the null hypothesis and conclude that the mean of fully vaccinated for hispanic people in Chelsea is significantly different from that in Wellesley.

### 3. White People

H0: μ1 = μ2
Ha: μ1 != μ2

stat = -21.231
p = 0.00000000

Conclusion: Since the p-value is small enough, we can reject the null hypothesis and conclude that the mean of fully vaccinated for white people in Chelsea is significantly different from that in Wellesley.

## Newton vs. Revere

### 1. Black People

H0: μ1 = μ2
Ha: μ1 != μ2

stat = 1.134
p = 0.25838889

Conclusion: Since the p-value is greater than the significance level, we fail to reject the null hypothesis and conclude that the mean of fully vaccinated for black people in Newton is NOT significantly different from that in Revere.

### 2. Hispanic People

H0: μ1 = μ2
Ha: μ1 != μ2

stat = -20.588
p = 0.00000000

Conclusion: Since the p-value is small enough, we can reject the null hypothesis and conclude that the mean of fully vaccinated for hispanic people in Newton is significantly different from that in Revere.

### 3. White People

H0: μ1 = μ2
Ha: μ1 != μ2

stat = 24.964
p = 0.00000000

Conclusion: Since the p-value is small enough, we can reject the null hypothesis and conclude that the mean of fully vaccinated for white people in Newton is significantly different from that in Revere.

## Newton vs. Wellesley

### 1. Black People

H0: μ1 = μ2
Ha: μ1 != μ2

stat = 29.140
p = 0.00000000

Conclusion: Since the p-value is small enough, we can reject the null hypothesis and conclude that the mean of fully vaccinated for black people in Newton is significantly different from that in Wellesley.

### 2. Hispanic People

H0: μ1 = μ2
Ha: μ1 != μ2

stat = 23.603
p = 0.00000000

Conclusion: Since the p-value is small enough, we can reject the null hypothesis and conclude that the mean of fully vaccinated for hispanic people in Newton is significantly different from that in Wellesley.

### 3. White People

H0: μ1 = μ2
Ha: μ1 != μ2

stat = 27.148
p = 0.00000000

Conclusion: Since the p-value is small enough, we can reject the null hypothesis and conclude that the mean of fully vaccinated for white people in Newton is significantly different from that in Wellesley.


## Revere vs. Wellesley

### 1. Black People

H0: μ1 = μ2
Ha: μ1 != μ2

stat = 24.957
p = 0.00000000

Conclusion: Since the p-value is small enough, we can reject the null hypothesis and conclude that the mean of fully vaccinated for black people in Revere is significantly different from that in Wellesley.

### 2. Hispanic People

H0: μ1 = μ2
Ha: μ1 != μ2

stat = 25.817
p = 0.00000000

Conclusion: Since the p-value is small enough, we can reject the null hypothesis and conclude that the mean of fully vaccinated for hispanic people in Revere is significantly different from that in Wellesley.

### 3. White People

H0: μ1 = μ2
Ha: μ1 != μ2

stat = 4.118
p = 0.00006067

Conclusion: Since the p-value is small enough, we can reject the null hypothesis and conclude that the mean of fully vaccinated for white people in Revere is significantly different from that in Wellesley.

In [71]:
# city = ['Chelsea', 'Newton', 'Revere', 'Wellesley']
# race = ['Black','Hispanic','White']
# def hypothesis_test(df):
# 	for i in range(len(city)):
# 		for j in range(len(city)):
# 			df1 = df[df['city'] == city[i]]
# 			df2 = df[df['city'] == city[j]]
# 			print(city[i], 'vs.', city[j])
# 			for a in range(len(race)):
# 				df_1 = df1[df1['race'] == race[a]]
# 				df_2 = df2[df2['race'] == race[a]]
# 				print(race[a])
# 				t_test(df_1['fully_vaccinated'], df_2['fully_vaccinated'])
#
# 			print('\n')
#
# hypothesis_test(df)

Chelsea vs. Chelsea
Black
stat=0.000, p=1.00000000
Probably the same distribution
Hispanic
stat=0.000, p=1.00000000
Probably the same distribution
White
stat=0.000, p=1.00000000
Probably the same distribution


Chelsea vs. Newton
Black
stat=-4.889, p=0.00000240
Probably different distributions
Hispanic
stat=20.504, p=0.00000000
Probably different distributions
White
stat=-36.057, p=0.00000000
Probably different distributions


Chelsea vs. Revere
Black
stat=-3.468, p=0.00067271
Probably different distributions
Hispanic
stat=1.391, p=0.16610254
Probably the same distribution
White
stat=-25.046, p=0.00000000
Probably different distributions


Chelsea vs. Wellesley
Black
stat=23.799, p=0.00000000
Probably different distributions
Hispanic
stat=25.192, p=0.00000000
Probably different distributions
White
stat=-21.231, p=0.00000000
Probably different distributions


Newton vs. Chelsea
Black
stat=4.889, p=0.00000240
Probably different distributions
Hispanic
stat=-20.504, p=0.00000000
Probably d

can you find the ranges for each of the time intervals that we are going to be working with - aka like the early covid vax rollout, mid vax rollout, end vax rollout, booster1 rollout, and booster2 rollout?
should just be a matter of running through the data and looking through the points


The Food and Drug Administration (FDA) first granted emergency use authorization to the Pfizer–BioNTech vaccine on December 10, 2020,[6] and mass vaccinations began four days later.

The Moderna vaccine was granted emergency use authorization on December 17, 2020,[7]

and the Janssen (Johnson & Johnson) vaccine was granted emergency use authorization on February 27, 2021.[8]

https://www.hhs.gov/coronavirus/covid-19-vaccines/index.html

without boosters: 3/9/21 - 12/14/21
early: 3/9/21 - 6/8/21
mid:6/15/21 - 9/7/21
end:9/7/21 - 12/14/21

booster1: 12/21/21 - 7/5/22
booster2: 7/12/22 - 10/4/22